src/injected/visualization/element-highlight.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { CSSProperties } from 'react';
+import * as React from 'react';
+
+import { NamedSFC } from '../../common/react/named-sfc';
+import { FeatureFlagStoreData } from '../../common/types/store-data/feature-flag-store-data';
+import { ClientUtils } from '../client-utils';
+import { RenderDialog } from '../dialog-renderer';
+import { HtmlElementAxeResults } from '../scanner-utils';
+import { DrawerUtils } from './drawer-utils';
+import { DrawerConfiguration, GetBoundingRect } from './formatter';
+import { HightlightBox, HightlightBoxDeps } from './highlight-box';
+
+export type ElementHighlightDeps = {
+    drawerUtils: DrawerUtils;
+    clientUtils: ClientUtils;
+} & HightlightBoxDeps;
+
+export interface ElementHighlightProps {
+    deps: ElementHighlightDeps;
+    element: Element;
+    elementResult: HtmlElementAxeResults;
+    bodyStyle: CSSStyleDeclaration;
+    docStyle: CSSStyleDeclaration;
+    drawerConfig: DrawerConfiguration;
+    featureFlagStoreData: FeatureFlagStoreData;
+    dialogRender: RenderDialog;
```

Comment:
should `dialogRender` and `getBoundingRect` be part of the deps instead?

Reply:
Unfortunately not; these are decided/created just prior to this component.

dialogRender uses data from the selected item.

getBoundingRect uses the element OR the drawerUtils IIRC.